parachain/pallets/basic-channel/src/outbound/mod.rs

```diff
@@ -181,18 +176,20 @@ pub mod pallet {
 					return Err(Error::<T>::Overflow.into())
 				}
 
-				<MessageQueue<T>>::append(Message {
+				<MessageQueue<T>>::try_append(Message {
 					target,
 					nonce: *nonce,
 					payload: payload.to_vec(),
-				});
-				Self::deposit_event(Event::MessageAccepted(*nonce));
-				Ok(())
+				})
+				.map_err(|_unit| Err(Error::<T>::QueueSizeLimitReached))
```

Comment:
I used match initially, but this is shorter. Which style should we prefer when handling results?

Reply:
We do not have a hard and fast rule about `map` vs `match`. I think the `map` is more readable in this case.